modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -175,37 +184,72 @@ public RabbitMQContainer withSSL(
         final MountableFile certFile,
         final MountableFile caFile,
         final SslVerification verify,
-        boolean failIfNoCert,
-        int verificationDepth
+        boolean failIfNoCert
     ) {
-        return withSSL(keyFile, certFile, caFile, verify, failIfNoCert)
-            .withEnv("RABBITMQ_SSL_DEPTH", String.valueOf(verificationDepth));
+        return withSSL(keyFile, certFile, caFile, verify, failIfNoCert, DEFAULT_SSL_VERIFICATION_DEPTH);
     }
 
     public RabbitMQContainer withSSL(
         final MountableFile keyFile,
         final MountableFile certFile,
         final MountableFile caFile,
-        final SslVerification verify,
-        boolean failIfNoCert
+        final SslVerification verify
     ) {
-        return withSSL(keyFile, certFile, caFile, verify)
-            .withEnv("RABBITMQ_SSL_FAIL_IF_NO_PEER_CERT", String.valueOf(failIfNoCert));
+        return withSSL(keyFile, certFile, caFile, verify, true);
     }
 
+    @SneakyThrows(IOException.class)
     public RabbitMQContainer withSSL(
         final MountableFile keyFile,
         final MountableFile certFile,
         final MountableFile caFile,
-        final SslVerification verify
+        final SslVerification verify,
+        boolean failIfNoCert,
+        int verificationDepth
     ) {
-        return withEnv("RABBITMQ_SSL_CACERTFILE", "/etc/rabbitmq/ca_cert.pem")
-            .withEnv("RABBITMQ_SSL_CERTFILE", "/etc/rabbitmq/rabbitmq_cert.pem")
-            .withEnv("RABBITMQ_SSL_KEYFILE", "/etc/rabbitmq/rabbitmq_key.pem")
-            .withEnv("RABBITMQ_SSL_VERIFY", verify.value)
-            .withCopyFileToContainer(certFile, "/etc/rabbitmq/rabbitmq_cert.pem")
+        Path advancedConfig = File.createTempFile("advanced-custom", ".config").toPath();
+        String advancedContent = createAdvancedConfigFileContent(
+            "/etc/rabbitmq/ca_cert.pem",
+            "/etc/rabbitmq/rabbitmq_cert.pem",
+            "/etc/rabbitmq/rabbitmq_key.pem",
+            verify,
+            failIfNoCert,
+            verificationDepth
+        );
+        Files.write(advancedConfig, advancedContent.getBytes(StandardCharsets.UTF_8));
+        return withCopyFileToContainer(certFile, "/etc/rabbitmq/rabbitmq_cert.pem")
             .withCopyFileToContainer(caFile, "/etc/rabbitmq/ca_cert.pem")
-            .withCopyFileToContainer(keyFile, "/etc/rabbitmq/rabbitmq_key.pem");
+            .withCopyFileToContainer(keyFile, "/etc/rabbitmq/rabbitmq_key.pem")
+            .withEnv("RABBITMQ_ADVANCED_CONFIG_FILE", "/etc/rabbitmq/advanced-custom.config")
```

Comment:
From what I saw and understand in this [comment](https://github.com/testcontainers/testcontainers-java/issues/5599#issuecomment-1185533946) I think it is a deliberated choice to use advanced configuration instead of configuration.
As RabbitMQContainer does not allow developers to define an advanced config file, this auto-generated file can't create a conflict. Generating a regular config file would override the one provided by the developer with `withRabbitMQConfig`

Reply:
oh, great! I forgot about those utility methods in RabbitMQContainer. Thanks for the hint!